draft-ietf-httpbis-priority.md

```diff
@@ -501,8 +501,68 @@ not specify the incremental(`i`) parameter.
+a particular order using priority. Expressing priority is therefore only a
+suggestion.
+
+A server can use priority signals along with other inputs to make scheduling
+decisions. No guidance is provided about how this can or should be done. Factors
+such as implementation choices or deployment environment also play a role. Any
+given connection is likely to have many dynamic permutations. For these reasons,
+there is no unilateral perfect scheduler and this document only provides some
+basic recommendations for implementations. 
+
+Clients cannot depend on particular treatment based on priority signals. Servers
+can use other information to prioritize responses.
+
+It is RECOMMENDED that, when possible, servers respect urgency values. Sending
+higher urgency responses before lower urgency responses.
+
+It is RECOMMENDED that, when possible, servers respect incremental values.
+Sending non-incremental responses of the same urgency one-by-one, and sending
+incremental responses of the same urgency in round-robin manner.
+
+It is RECOMMENDED that a server considers request generation order as an input
+to prioritization. Prioritizing concurrent requests at the same urgency level
+based on the Stream ID, which corresponds to the order in which clients make
+requests, ensures that clients can use request ordering to influence response
+order.
+
+The incremental parameter ({{incremental}}) indicates how a client processes
+response bytes as they arrive. Non-incremental resources are only used when all
+of the response payload has been received. Incremental resources are used as
+parts, or chunks, of the response payload are received. Therefore, the timing of
+response data reception at the client, such as the time to early bytes or the
+time to receive the entire payload, plays an important role in perceived
+performance. Timings depend on resource size but this scheme provides no
+explicit guidance about how a server should use size as an input to
+prioritization. Instead, the follow examples demonstrate how a server that
+strictly abides the scheduling guidance based on urgency and request generation
+order could find that early requests prevent serving of later requests.
+
+1. At the same urgency level, a non-incremental request for a large resource
+   followed by an incremental request for a small resource.
+2. At the same urgency level, an incremental request of indeterminate length
+   followed by a non-incremental large resource.
+
+It is RECOMMENDED that servers avoid such starvation where possible. The method
+to do so is an implementation decision. For example, a server might
+pre-emptively send responses of a particular incremental type based on other
+information such as content size.
+
+An HTTP/2 server implementing the Extensible Priorities scheme instead of the
+HTTP/2 priority sends SETTINGS_DEPRECATE_HTTP2_PRIORITIES; see {{disabling}}. It
+SHOULD NOT act on priority signals belonging to the HTTP/2 scheme. The absence
```

Comment:
This sentence sounds like that a server cannot respect the PRIORITY frames sent by a legacy HTTP/2 client.

I think that the intent is to state something like: When a client sends SETTINGS_DEPRECATE_HTTP2_PRIORITIES, a server SHOULD NOT act ...

Reply:
Ok I think we come to this from different viewpoints. I was trying to address the statement in {{disabling}}

> The SETTINGS frame precedes any priority signal sent from a client in HTTP/2, so a server can determine if it should respect the HTTP/2 scheme before building state.

I've reworked the paragraph to accomodate either client or server sending the setting, PTAL.